cli/src/commands/containers.rs

```diff
@@ -742,3 +779,68 @@ pub async fn start_postgres(docker: &Docker, container_id: &str) -> anyhow::Resu
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::control_file::ControlFile;
+
+    use super::{validate_extension_files, ExtensionFiles};
+
+    #[test]
+    fn validates_extension_files() {
+        // Valid: Contains files of the three kinds
+        validate_extension_files(&ExtensionFiles {
+            sharedir: vec![
+                "pg_stat_statements--1.0--1.1.sql".into(),
+                "pg_stat_statements--1.1--1.2.sql".into(),
+            ],
+            pkglibdir: vec!["pg_stat_statements.so".into()],
+            control_file: Some(ControlFile {
+                directory: None,
+                module_pathname: Some("'$libdir/pg_stat_statements'".into()),
+                requires: None,
+            }),
+        })
+        .unwrap();
+
+        // Valid: Extension with a control file and SQL files, but no shared objects
+        validate_extension_files(&ExtensionFiles {
+            sharedir: vec!["pgmq--1.4.4--1.4.5.sql".into(), "pgmq--1.4.5.sql".into()],
+            pkglibdir: vec![],
+            control_file: Some(ControlFile {
+                directory: None,
+                module_pathname: Some("'$libdir/pgmq'".into()),
+                requires: None,
+            }),
+        })
+        .unwrap();
+
+        // Valid: Extension with a shared object but no control file
+        validate_extension_files(&ExtensionFiles {
+            sharedir: vec![],
+            pkglibdir: vec!["auth_delay.so".into()],
+            control_file: None,
+        })
+        .unwrap();
+
+        // Invalid: No control file and no SQL file
```

Comment:
```suggestion
        // Invalid: Control file and no SQL file
```

Should also test for the inverse: SQL files but no control file.